openapi-cli/src/main/java/io/ballerina/openapi/cmd/BallerinaCodeGenerator.java

```diff
@@ -154,10 +142,35 @@ public void generateClientAndService(String definitionPath, String serviceName,
         //Update type definition list
         List<TypeDefinitionNode> preGeneratedTypeDefNodes = new ArrayList<>(
                 clientGenerator.getBallerinaAuthConfigGenerator().getAuthRelatedTypeDefinitionNodes());
-        List<TypeDefinitionNode> typeInclusionRecords = serviceGenerator.getTypeInclusionRecords();
         List<TypeDefinitionNode> typeDefinitionNodeList = clientGenerator.getTypeDefinitionNodeList();
-        preGeneratedTypeDefNodes.addAll(typeInclusionRecords);
         preGeneratedTypeDefNodes.addAll(typeDefinitionNodeList);
+        String serviceContent = "";
+        if (complexPaths.isEmpty()) {
+            OASServiceMetadata oasServiceMetadata = new OASServiceMetadata.Builder()
+                    .withOpenAPI(openAPIDef)
+                    .withFilters(filter)
```

Comment:
`withFilters` name implies that this can method should take a list of filters. But here we are only passing one filter. What is the reason for using such naming convention ?

Reply:
We use filters since currently we have two filters `tags`, and `operationId` therefore we used to intend the filters. We will address this within this ongoing revamp https://github.com/ballerina-platform/ballerina-library/issues/6009

Reply:
`filters` noun means multiple filters, not a filter of one of the two types. We should be considerate about the naming conventions we adopt. Lets make sure that this is fixed in the revamp effort.